src/librustc_codegen_llvm/back/write.rs

```diff
@@ -575,6 +582,22 @@ unsafe fn optimize(cgcx: &CodegenContext,
                                            `{}` but LLVM does not \
                                            recognize it", pass));
             }
+            if pass == "name-anon-globals" {
+                have_name_anon_globals_pass = true;
+            }
+        }
+
+        if using_thin_buffers && !have_name_anon_globals_pass {
+            // As described above, this will probably cause an error in LLVM
+            if config.no_prepopulate_passes {
+                diag_handler.err("The current compilation is going to use thin LTO buffers without \
```

Comment:
I initially tried to make this a warning as suggested on IRC. That however proved to be very difficult, since warnings issued during codegen don't seem to be displayed until after codegen completes. In this case, that would mean that an LLVM crash would occur before the warning about the crash is displayed, meaning it won't be displayed at all.

Reply:
Weird, but an error is fine as well at this point in code.